### PR TITLE
Add WAL file archiving

### PR DESCRIPTION
**Pull request by joshraker**
Reviving #69 with changes to enable it on new databases by default but leaving existing databases unchanged.

https://aptible.zendesk.com/agent/tickets/25047

### CODE REVIEW

**Comment by joshraker** [comment]
`ALTER SYSTEM` was introduced in PG 9.4 so 9.3 doesn't have any notion of this or the `postgresql.auto.conf` file which is why the tests are failing. I'm just going to disable archiving for 9.3 and skip the tests for this version since it's been EOL for a year and a half.

**Comment by almathew** [comment]
@joshraker You'll probably also want this so we're actually pulling in the pagerduty key: https://github.com/aptible/sweetness/pull/1065/files

That PR isn't 100% mergable as-is, since the variable we use to store the key changed: https://github.com/aptible/sweetness/blob/master/lib/sweetness/stack_context.rb#L636

It shouldn't be deployed before this is merged, though, since that would result in the pagerduty key never getting removed from the environment variables

File: bin/run-database.sh [hunk 1]
```diff
@@ -171,10 +195,14 @@ elif [[ "$1" == "--restore" ]]; then
...
 elif [[ "$1" == "--readonly" ]]; then
   pg_init_conf
+  pg_init_pagerduty_notify
+  pg_init_archive
   pg_run_server --default_transaction_read_only=on
 
 else
   pg_init_conf
+  pg_init_pagerduty_notify
+  pg_init_archive
```
Comment (joshraker) [inline bin/run-database.sh:205]: Is `pg_init_archive` included here so existing databases will be able to use WAL archiving or is there another reason?

Comment (almathew) [reply bin/run-database.sh:205]: Yes, that's the motivation for having it in both places

**Comment by almathew** [comment]
This looks fine, but before I approve, has anyone done any manual testing of this (other than me, obviously)? If not, I'm happy to run through testing with someone quick. I'm just a little nervous about merging otherwise, since we can't run this through aptible-integration first, for example.

**Comment by joshraker** [comment]
Manually testing PITR or running the image locally and seeing how it handles archiving?

**Comment by joshraker** [comment]
I've messed around with archiving locally by setting `wal_keep_segments` to `0` and starting and stopping the database. It's functioning as I would expect.

```
root@4d19bb6ce72d:/# ls -lA /var/archive/
total 49152
-rw------- 1 postgres postgres 16777216 Apr 17 21:14 000000010000000000000001
-rw------- 1 postgres postgres 16777216 Apr 17 21:16 000000010000000000000002
-rw------- 1 postgres postgres 16777216 Apr 17 21:16 000000010000000000000003
root@4d19bb6ce72d:/# ls -lA /var/db/pg_wal/
total 49156
-rw------- 1 postgres postgres 16777216 Apr 17 21:16 000000010000000000000003
-rw------- 1 postgres postgres 16777216 Apr 17 21:16 000000010000000000000004
-rw------- 1 postgres postgres 16777216 Apr 17 21:16 000000010000000000000005
drwx------ 2 postgres postgres     4096 Apr 17 21:16 archive_status
```

**Comment by almathew** [comment]
All of the above: confirming that we can switch an existing database ok, confirming that archiving works, confirming that the restore process makes sense

**Comment by almathew** [comment]
Stale PR cleanup 😢

**Pull request closed without merging**